src/application.coffee

```diff
@@ -613,14 +621,16 @@ application.update = update = (force) ->
 								fetch(app) if needsDownload
 							.then ->
 								start(app)
-						else if _.includes(toBeUpdated, appId)
+						else if _.includes(toBeUpdated, appId) or _.includes(toBeRestarted, appId)
 							app = remoteApps[appId]
 							conf = JSON.parse(app.config)
 							forceThisApp =
 								conf['RESIN_SUPERVISOR_OVERRIDE_LOCK'] == '1' ||
 								conf['RESIN_OVERRIDE_LOCK'] == '1'
 							strategy = conf['RESIN_SUPERVISOR_UPDATE_STRATEGY']
 							timeout = conf['RESIN_SUPERVISOR_HANDOVER_TIMEOUT']
+							if _.includes(toBeRestarted, appId) and strategy is 'delete-then-download'
+								strategy = 'download-then-kill'
```

Comment:
Just realized this should actually be fixed in the strategy itself, as it also affects updates that don't require a download.